### PR TITLE
Tapenames elasticarray

### PR DESCRIPTION
**Pull request by gperciva**

### CODE REVIEW

**Comment by gperciva** [comment]
This another one split from #261.  Not certain if I should rebase the history or not, so I left it undone.

File: tar/bsdtar.c [hunk 1]
```diff
@@ -783,14 +790,15 @@ main(int argc, char **argv)
 	 * few bytes off anyway since the command line, including "--dry-run"
 	 * is included in the metadata.
 	 */
-	if (bsdtar->option_dryrun && (bsdtar->ntapes == 0)) {
+	if (bsdtar->option_dryrun &&
+	    (strlist_getsize(bsdtar->tapenames_setup) == 0)) {
```
Comment (gperciva) [inline tar/bsdtar.c]: This is a bit ugly, but if we're not incrementing `bsdtar->ntapes`, I guess we don't have an alternative.

**Comment by cperciva** [comment]
The final result looks good.  I think this can probably just be a single commit...

**Comment by gperciva** [comment]
Updated PR with rebased history.  Yes, I intended it to be a single commit.

**Pull request merged**